src/translate_c/ast.zig

```diff
@@ -1253,6 +1259,13 @@ fn renderNode(c: *Context, node: Node) Allocator.Error!NodeIndex {
                 },
             });
         },
+        .this => {
+            return renderBuiltinCall(c, "@This", &.{});
+        },
+        .field => {
+            const payload = node.castTag(.field).?.data;
+            return renderBuiltinCall(c, "@field", &.{ payload.lhs, payload.rhs });
```

Comment:
I get an error using bare `@"_"` on the RHS:

```zig
pub var @"_": c_int = 5;
pub fn main() void {
    var x: c_int = @"_";
    _ = x;
}
```

```
./test.zig:3:20: error: `_` may only be used to assign things to
    var x: c_int = @"_";
                   ^
```

It's only relevant when trying to assign a global `_` variable to something - local `_`'s already get mangled, so it seems pretty edge-case anyway. 

Is there a way to get the current file's struct regardless of current scope? Alternatively could put something like `const @"This" = @This();` at the top of the file.

Reply:
> I get an error using bare @"_" on the RHS:

That is a stage1 limitation, it could probably be fixed easily by just removing the check from stage1 completely since `ast-check` implements the it properly.

> Is there a way to get the current file's struct regardless of current scope?

Nope, I had to put a `const file_struct = @This();` in `type.zig` for the same reason.

Reply:
Based on some quick testing this patch should fix the problem:
```diff
diff --git a/src/stage1/astgen.cpp b/src/stage1/astgen.cpp
index e3eebab0b..876dd384a 100644
--- a/src/stage1/astgen.cpp
+++ b/src/stage1/astgen.cpp
@@ -3821,9 +3821,6 @@ static IrInstSrc *astgen_identifier(Stage1AstGen *ag, Scope *scope, AstNode *nod
             const_instruction->value->special = ConstValSpecialStatic;
             const_instruction->value->data.x_ptr.special = ConstPtrSpecialDiscard;
             return &const_instruction->base;
-        } else {
-            add_node_error(ag->codegen, node, buf_sprintf("`_` may only be used to assign things to"));
-            return ag->codegen->invalid_inst_src;
         }
     }
 
```

Reply:
Ok, I removed `@This()` and `@field()` from the translate-c AST and also removed the stage1 check against assigning from `_`. With that removed it looks like I can just do `var foo = @"_";`

Reply:
Testing out this branch some more, I noticed that with the underscore change, I can initialize and assign from `@"_"` but cannot assign *to* it:

```zig
const std = @import("std");
var @"_": u32 = 5;
test "underscore" {
    try std.testing.expectEqual(@as(u32, 5), @"_");
    var x = @"_";
    try std.testing.expectEqual(@as(u32, 5), x);
    @"_" = 6;
    try std.testing.expectEqual(@as(u32, 6), @"_");
}
```

```
expected 6, found 5
test "underscore"... FAIL (TestExpectedEqual)
/Users/ehaas/source/zig/lib/std/testing.zig:82:17: 0x108266d6b in std.testing.expectEqual (test)
                return error.TestExpectedEqual;
                ^
/Users/ehaas/test.zig:8:5: 0x108264b03 in test "underscore" (test)
    try std.testing.expectEqual(@as(u32, 6), @"_");
    ^
```

Reply:
I think it's fine to treat this as yet another stage1 miscompilation since it is non breaking. We'll soon enough have stage2 ready which will handle this correctly.